python/cudf/cudf/pylibcudf_tests/test_string_replace.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+
+
+@pytest.fixture(scope="module")
+def pa_data_col():
+    return pa.array(["a", "c", "A", "aa", None, "aaaaaaaaa", "AAAA", "ÁÁÁÁ"])
+
+
+@pytest.fixture(scope="module")
+def plc_data_col(pa_data_col):
+    return plc.interop.from_arrow(pa_data_col)
+
+
+@pytest.fixture(scope="module", params=["a", "c", "A", "Á", "aa", "ÁÁÁ"])
+def scalar_repl_target(request):
+    pa_target = pa.scalar(request.param, type=pa.string())
+    return (request.param, plc.interop.from_arrow(pa_target))
+
+
+@pytest.fixture(scope="module", params=["b", "B", "", "B́"])
+def scalar_repl(request):
+    pa_repl = pa.scalar(request.param, type=pa.string())
+    return (request.param, plc.interop.from_arrow(pa_repl))
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        ["a", "c", "A", "ÁÁÁÁ"],
+    ],
+)
+def col_repl_target(request):
+    pa_target = pa.array(request.param, type=pa.string())
+    return (pa_target, plc.interop.from_arrow(pa_target))
+
+
+@pytest.fixture(
+    scope="module",
+    params=[
+        [
+            "",
+            "z",
+            "XX",
+            "blahblah",
+        ]
+    ],
+)
+def col_repl(request):
+    pa_repl = pa.array(request.param, type=pa.string())
+    return (pa_repl, plc.interop.from_arrow(pa_repl))
+
+
+@pytest.mark.parametrize("maxrepl", [-1, 1, 2, 10])
+def test_replace(
+    pa_data_col, plc_data_col, scalar_repl_target, scalar_repl, maxrepl
+):
+    pa_target, plc_target = scalar_repl_target
+    pa_repl, plc_repl = scalar_repl
+    got = plc.strings.replace.replace(
+        plc_data_col, plc_target, plc_repl, maxrepl
+    )
+
+    expected = pa.compute.replace_substring(
+        pa_data_col,
+        pattern=pa_target,
+        replacement=pa_repl,
+        max_replacements=maxrepl,
+    )
+
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize("startstop", [(0, -1), (0, 0), (1, 3)])
+def test_replace_slice(pa_data_col, plc_data_col, scalar_repl, startstop):
+    pa_repl, plc_repl = scalar_repl
+    start, stop = startstop
+    got = plc.strings.replace.replace_slice(
+        plc_data_col, plc_repl, start, stop
+    )
+
+    if stop == -1:
+        # pyarrow doesn't support -1 as stop, so just set to really big number
+        stop = 1000
```

Comment:
To be more precise, could we add a TODO here to switch to using `count_characters() + 1` once we've exposed that libcudf function?